src/modules/chat_highlight_blacklist_keywords/index.js

```diff
@@ -144,20 +144,52 @@ class ChatHighlightBlacklistKeywordsModule {
     this.handleHighlightSound = this.handleHighlightSound.bind(this);
   }
 
-  updateDefaultHighlightKeywords() {
-    const value = settings.get(SettingIds.HIGHLIGHT_KEYWORDS);
-    if (Object.keys(value).length > 0) return;
+  readRepairKeywords() {
+    let highlightKeywordsValue = settings.get(SettingIds.HIGHLIGHT_KEYWORDS);
+    const blacklistKeywordsValue = settings.get(SettingIds.BLACKLIST_KEYWORDS);
+
+    highlightKeywordsValue = this.updateDefaultHighlightKeywords(highlightKeywords) || {};
+
+    for (const keywordsValue of [highlightKeywordsValue, blacklistKeywordsValue]) {
+      let updated = false;
+
+      for (const value of Object.values(keywordsValue)) {
+        if (![KeywordTypes.EXACT, KeywordTypes.WILDCARD].includes(value.type)) {
+          continue;
+        }
+
+        if (value.type === KeywordTypes.EXACT) {
+          value.keyword = `<${value.keyword}>`;
+        }
+        value.type = KeywordTypes.MESSAGE;
+        updated = true;
+      }
+
+      if (updated) {
+        settings.set(
+          keywordsValue === highlightKeywordsValue ? SettingIds.HIGHLIGHT_KEYWORDS : SettingIds.BLACKLIST_KEYWORDS,
+          keywordsValue
+        );
+      }
+    }
+  }
+
+  updateDefaultHighlightKeywords(value) {
+    if (value != null) return null;
     const user = twitch.getCurrentUser();
-    if (user == null) return;
+    if (user == null) return null;
```

Comment:
should this return empty object if it's not null?

Reply:
`  updateDefaultHighlightKeywords() {
    const value = settings.get(SettingIds.HIGHLIGHT_KEYWORDS);
    if (value != null) return;
    const user = twitch.getCurrentUser();
    if (user == null) return;

    settings.set(SettingIds.HIGHLIGHT_KEYWORDS, {
      0: {
        id: 0,
        type: KeywordTypes.MESSAGE,
        status: null,
        keyword: user.name,
      },
    });
  }`

I think alternatively we should keep the function like so, we shouldn't build it around another function. granted this is ineffecient on the first time a logged user uses the extension but afterwards there won't be as many reads and writes